### PR TITLE
Ft validation

### PR DESCRIPTION
**Pull request by unadine**
What this PR do
-firebase configuration
-setup connection to firestore
-validate inputs
-sending messages
-commenting on blogs and viewing comments
-CRUD to the admin panel
-Geolocation API
-Authentication

### CODE REVIEW

File: Blogs/js/blog-details.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
+let id=location.hash.slice(1);
+
+function renderArticle(article){
```
Comment (harerakalex) [inline Blogs/js/blog-details.js]: Nice job @unadine, Good variable naming, however, you should use es6 to make your code more modern and more readable.

File: Blogs/js/blog-details.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+
+
+
+
+
+
+
+
+// let id=location.hash.slice(1);
```
Comment (harerakalex) [inline Blogs/js/blog-details.js]: Better remove the unnecessary comments from your code

File: Blogs/js/blogs1.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+
+
+
+
+
+
+
+
+// const msg_section= document.querySelector('#message-list1');
```
Comment (harerakalex) [inline Blogs/js/blogs1.js]: Remove this code instead of commenting them

File: js/authentication.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    const email = signinForm['signin-email'].value;
+    const password = signinForm['signin-password'].value;
+
+    //signin users
+    auth.signInWithEmailAndPassword(email, password).then(cred=>{
+        console.log(cred)
+        window.location.href="../MYBRAND/Admin/create.html";
+        signinForm.reset();
+    }).catch(error => {
+        switch (error.code) {
```
Comment (harerakalex) [inline js/authentication.js:24]: Great idea for using  `Switch case`.

File: js/scripts.js [hunk 5]
```diff
@@ -1,8 +1,41 @@
...
 }
-  
+// sign in
+const modal = document.getElementById('signin');
+window.onclick = function(event) {
+    if (event.target == modal) {
+        modal.style.display = "none";
+    }
+}
+// log in
```
Comment (harerakalex) [inline js/scripts.js]: The same here, remove it

File: js/scripts.js [hunk 6]
```diff
@@ -1,8 +1,41 @@
...
+//     if (event.target == modal) {
+//         modal.style.display = "none";
+//     }
+// }
+//subscription
+const secondform= document.getElementById('secondform');
+secondform.addEventListener('submit',(e)=>{
+  e.preventDefault();
+  const secondemail = document.getElementById('secondemail').value;
+   var pattern = /^[^ ]+@[^ ]+\.[a-z]{2,3}$/;
```
Comment (harerakalex) [inline js/scripts.js:21]: Wow, Nice, I like that you used regex

**Review by harerakalex — CHANGES_REQUESTED** [review]
Good job @unadine, kindly check out the below comment

**Comment by James-nyagatare** [comment]
Great job @unadine  apart from those uneccessary comments everything looks good for me.

**Review by harerakalex — APPROVED** [review]

**Pull request merged**